src/main/java/com/example/dao/ProductDAOImpl.java

```
@@ -0,0 +1,315 @@
+package com.example.dao;
+
+import com.example.model.Product;
+import com.example.util.HibernateUtil;
+import org.hibernate.Session;
+
+import javax.persistence.Query;
+import javax.persistence.criteria.CriteriaBuilder;
+import javax.persistence.criteria.CriteriaQuery;
+import javax.persistence.criteria.Root;
+import java.util.List;
+
+public class ProductDAOImpl extends AbstractDAO<Product, Object> {
+
+    public Product findNewProduct() {
+
+        try {
+            Session session = HibernateUtil.getSesstionFactory().openSession();
+
+            CriteriaBuilder builder = session.getCriteriaBuilder();
+            CriteriaQuery<Product> query = builder.createQuery(Product.class);
+            Root<Product> root = query.from(Product.class);
+
+            query.select(root).orderBy(builder.desc(root.get("id")));
+
+            return session.createQuery(query).setMaxResults(1).getSingleResult();
+
+        } catch (Exception ex) {
+            ex.printStackTrace();
```

I think if we have exception we should return the error for the user.